### DISCUSSION TITLE
Undestand how the invoices base price and VAT calculation is working

### DISCUSSION THREAD

**Discussion by alxsbn** [PrestaShop/PrestaShop#36634] [Q&A]
Hello everyone,

When I'm using data from `ps_orders` it's pretty easy to extract taxes. But I'm not abe to reproduce the exact same values using `ps_order_detail` and I need to extract VAT details (per country per rate). 

I already see some issues with rounding but that's globally OK.

No my main issue here is to understand the whole calculation, especially those included in the invoices. Below you will find one strange example with the tax tab including negative taxes. But surprise, the total taxes of the tax tab is equal to the global tax. I want to understand how is this possible?
![image](https://github.com/user-attachments/assets/69823f2b-c7c0-4b94-9476-0662c318e60b)

But more globally Can someone explain me how the base price is calculated, especially using discounts.

Thanks in advance

**Comment by Hlavtox** [comment]
Hi, please open a bug report, this is a serious bug. :-)
